File: src/main/java/com/polygloat/dtos/request/SignUpDto.java
package com.polygloat.dtos.request;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.validator.constraints.Length;

import javax.validation.constraints.Email;
import javax.validation.constraints.NotBlank;

@Data
@Builder
@AllArgsConstructor
@NoArgsConstructor
public class SignUpDto {
    @NotBlank
    String name;

    @Email
    @NotBlank
    String email;

    @Length(min = 8, max = 100)
    @NotBlank
    String password;

    String invitationCode;
}

File: src/main/java/io/polygloat/dtos/request/SignUpDto.kt
package io.polygloat.dtos.request

import javax.validation.constraints.Email
import javax.validation.constraints.NotBlank
import javax.validation.constraints.Size

data class SignUpDto(
        @NotBlank
        var name: String? = null,

        @Email @NotBlank
        var email: String? = null,

        @Size(min = 8, max = 100)
        @NotBlank
        var password: String? = null,

        var invitationCode: String? = null,

        var callbackUrl: String? = null
)